src/js/sjcl-custom/sjcl-ecdsa-signdeterministic.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+/*global sjcl*/
+/*eslint-disable new-cap*/
+
+sjcl.ecc.ecdsa.secretKey.prototype.generateK =
+function (hash, hashObject, q) {
+
+  hashObject = hashObject || sjcl.hash.sha256;
+  var secret = this.get();
+
+  /* k = 0x00 0x00 0x00 ... 0x00 */
+  var k = [];
+  while (sjcl.bitArray.bitLength(k) < sjcl.bitArray.bitLength(hash)) {
+    k = sjcl.bitArray._shiftRight(k, 8);
```

Comment:
You can probably do it by just doing an array fill of `(qlen+7)/8` bytes, then using `sjcl.codec.bytes.toBits`. Currently this routine (`generateK`) doesn't seem to work. You might have more luck decomposing it into various steps/routines that are testable. You could possibly even implement a class.

bitArray's are a bit funny, with the logical unit being a word, rather than byte. So you need to be careful. If you log out `sjcl.codec.bytes.fromBits(v)` you'll see that the first 3 octets are actually `0x0`, even though there's 32 octets. It's probably the `bitLength(...)` method biting you.